src/test/java/org/folio/rest/api/InstanceStorageTest.java

```diff
@@ -475,6 +475,50 @@ public void canGetAllInstances()
       hasItem(identifierMatches(UUID_ASIN.toString(), "B01D1PLMDO")));
   }
 
+  @Test
+  public void canSearchByClassificationNumber()
+    throws MalformedURLException,
+    InterruptedException,
+    ExecutionException,
+    TimeoutException {
+    
+    UUID firstInstanceId = UUID.randomUUID();
+    UUID secondInstanceId = UUID.randomUUID();
+    UUID classificationTypeId = UUID.randomUUID();
+
+    JsonObject firstClassifications = new JsonObject();
+    firstClassifications.put("classificationTypeId", classificationTypeId.toString());
+    firstClassifications.put("classificationNumber", "K1 .M385");
+
+    JsonObject secondClassifications = new JsonObject();
+    secondClassifications.put("classificationTypeId", classificationTypeId.toString());
+    secondClassifications.put("classificationNumber", "KB1 .A437");
+
+    JsonObject firstInstanceToCreate = smallAngryPlanet(firstInstanceId)
+    .put("classifications", new JsonArray()
+    .add(firstClassifications));
+    JsonObject secondInstanceToCreate = nod(secondInstanceId)
+    .put("classifications", new JsonArray()
+    .add(secondClassifications));
+
+    createInstance(firstInstanceToCreate);
+    createInstance(secondInstanceToCreate);
+
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
+    
+    client.get(instancesStorageUrl("?query=classifications%3D%22K1%20.M385%22"), StorageTestSuite.TENANT_ID,
+    ResponseHandler.json(getCompleted));
+
+    Response response = getCompleted.get(5, TimeUnit.SECONDS);
+
+    JsonObject responseBody = response.getJson();
+
+    JsonArray allInstances = responseBody.getJsonArray("instances");
```

Comment:
```suggestion
    JsonArray allInstances = searchForInstances("classifications =/@classificationNumber \"K1 .M385\"");
```

This checks that the arraySubfield has been declared in schema.json. If the declaration is missing or wrong the /@ will fail.

Reply:
Thanks @julianladisch 

I agree with using `searchForInstances` to avoid the duplicated code.

@K-Felk please define separate tests for the separate CQL variants.